gensim/models/word2vec.py

```diff
@@ -776,6 +795,9 @@ def job_producer():
                             progress = 1.0 * pushed_words / total_words
                         next_alpha = self.alpha - (self.alpha - self.min_alpha) * progress
                         next_alpha = max(self.min_alpha, next_alpha)
+                        # raise warning in case alpha rises again
+                        if next_alpha > current_alpha:
+                            warnings.warn("'alpha' risen again after descending.")
```

Comment:
I don't see any way this warning could ever be triggered, given `current_alpha` is a local set above. (Have you been able to trigger it in your testing?) The user potential-error we're trying to warn about is when someone, in repeated `train()`s, somehow starts again at a higher `alpha`. To catch that will require remembering a sort of lowest-alpha-yet-reached, across multiple calls to `train()`.